File: php/Program/ProdukElektronik.php
<?php
// =============================
// Class ProdukElektronik
// Level 1 (Parent class)
// =============================
class ProdukElektronik {
    private $namaProduk;  // Nama produk elektronik
    private $merek;       // Merek produk
    private $harga;       // Harga produk
    private $gambar;      // File gambar produk

    // =============================
    // Constructor dengan parameter opsional
    // =============================
    public function __construct($namaProduk="", $merek="", $harga=0, $gambar=""){
        $this->namaProduk = $namaProduk;
        $this->merek = $merek;
        $this->harga = $harga;
        $this->gambar = $gambar;
    }

    // =============================
    // Getter & Setter
    // =============================
    public function getNamaProduk() { return $this->namaProduk; }
    public function setNamaProduk($namaProduk) { $this->namaProduk = $namaProduk; }

    public function getMerek() { return $this->merek; }
    public function setMerek($merek) { $this->merek = $merek; }

    public function getHarga() { return $this->harga; }
    public function setHarga($harga) { $this->harga = $harga; }

    public function getGambar() { return $this->gambar; }
    public function setGambar($gambar) { $this->gambar = $gambar; }
}
?>

File: cpp/Program/ProdukElektronik.cpp
#include <bits/stdc++.h>   // Mengimpor seluruh header standar C++ (praktis, tapi tidak direkomendasikan di production)
using namespace std;

// ===== Class ProdukElektronik =====
// Level paling dasar (base class) yang merepresentasikan produk elektronik umum
class ProdukElektronik {
private:
    string namaProduk;  // Nama produk elektronik (contoh: Lampu LED, TV, dll)
    string merek;       // Nama merek produk (contoh: Philips, LG, Samsung)
    int harga;          // Harga produk dalam satuan rupiah

public:
    // Constructor kosong: bisa dipakai saat membuat objek tanpa langsung mengisi atribut
    ProdukElektronik() {}

    // Constructor dengan parameter: langsung mengisi atribut produk
    ProdukElektronik(string namaProduk, string merek, int harga) {
        this->namaProduk = namaProduk;
        this->merek = merek;
        this->harga = harga;
    }

    // Getter: untuk mengambil nilai atribut (membaca data)
    string getNamaProduk() { return namaProduk; }
    string getMerek() { return merek; }
    int getHarga() { return harga; }

    // Setter: untuk mengubah nilai atribut (menulis data)
    void setNamaProduk(string namaProduk) { this->namaProduk = namaProduk; }
    void setMerek(string merek) { this->merek = merek; }
    void setHarga(int harga) { this->harga = harga; }

    // Destructor: dipanggil otomatis saat objek dihapus (tidak ada aksi khusus di sini)
    ~ProdukElektronik() {}
};